src/main/java/Wessy.java

```diff
@@ -0,0 +1,279 @@
+        }
+        if (userInput.charAt(idx - 1) != ' ') {
+            throw new MissingSpacingException(keyword, false);
+        }
+        if (userInput.length() == idx + keyword.length() ||
+                userInput.charAt(idx + keyword.length()) != ' ') {
+            throw new MissingSpacingException(keyword, true);
+        }
+    }
+
+    static void checkForEmptyList(CmdType type) throws EmptyListException {
+        if (tasks.isEmpty()) {
+            throw new EmptyListException(type.toString());
+        }
+    }
+
+    static void printNormal(String... linesOfString) {
+        System.out.println(OPENING_LINE);
+        for (String line : linesOfString) {
+            println(line);
+        }
+        System.out.println(CLOSING_LINE);
+    }
+
+    static void printAdded(Task task) {
+        int size = tasks.size();
+        String numOfTasks = " " + size + " task";
+        if (size > 1) {
+            numOfTasks += "s";
+        }
+        printNormal("Got it. I've added this task:",
+                "  " + task, "Now you have" + numOfTasks + " in the list.");
+    }
+
+    static void println(String str) {
+        int length = str.length();
+        String message = "   |   " + str;
+        for (int i = 0; i < 70 - length - 3; i++) {
+            message += " ";
+        }
+        message += "|";
+        System.out.println(message);
+    }
+
+    static Task add(String[] strings) {
+        int len = strings.length;
+        if (len == 1) {
+            tasks.add(new ToDo(strings[0]));
+        } else if (len == 2) {
+            tasks.add(new Deadline(strings[0], strings[1]));
+        } else if (len == 3) {
+            tasks.add(new Event(strings[0], strings[1], strings[2]));
+        }
+        return tasks.get(tasks.size() - 1);
+    }
+
+    static void printList() {
+        int size = tasks.size();
+        System.out.println(OPENING_LINE);
+        if (size == 0) {
+            println("WOOHOO! You do not have any task on the list.");
+        } else {
+            println("Here are the tasks in your list:");
+            for (int i = 0; i < size; i++) {
+                println((i + 1) + "." + tasks.get(i));
+            }
+        }
+        System.out.println(CLOSING_LINE);
+    }
+
+    static int parseInt(String userInput, CmdType type) throws
+            EmptyListException, MissingInputException,
+            MissingSpacingException, NumberFormatException,
+            ArrayIndexOutOfBoundsException, UnspecifiedTimeException {
+        checkForEmptyList(type);
+        checkForMissingInput(userInput, type);
+        checkForSpacingAftCmd(userInput, type);
+        int idx = Integer.parseInt(userInput.substring(
+                type.len() + 1)) - 1;
+        if (idx < 0 || idx >= tasks.size()) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return idx;
+    }
+
+    static void markOrUnmark(String userInput, boolean isMark) throws
```

Comment:
Maybe these functions can also be abstracted away into a mark event class for example?